flasher.h

```diff
@@ -39,6 +39,7 @@
 #include <QFile>
```

Comment:
In flasher.h file, we can use forward declaration for SocketClient, so we do not need to include socket client header file here (it will be included in flasher source file). And, socket_client_ should be shared_ptr.

namespace socket {

class SocketClient;

} // namespace socket

Also, we need to include QJsonArray.